docs/blog/articles/Enhancing-func-cli-ux.md

```diff
@@ -0,0 +1,94 @@
+# My LFX Mentorship Experience: Enhancing the Knative func CLI Experience
+
+**Author: [Rayyan Seliya](https://www.linkedin.com/in/rayyan-seliya/){:target="_blank"}, LFX'25 Mentee**
+
+Over the past three months, I've had the incredible opportunity to participate in the LFX Mentorship Program. My specific project was titled [CNCF - Knative: Enhancing the Knative func CLI Experience](https://github.com/knative/ux/issues/196){:target="_blank"}. As the term wraps up, I want to take a moment to reflect on my journey and share it with you.
+
+## Project Overview
+
+Knative Functions enables developers to easily create, build, and deploy stateless, event driven functions as Knative Services by using the `func` CLI. The goal of this LFX project was to conduct a comprehensive UX evaluation of the func CLI, focusing on identifying usability issues, understanding developer workflows, and gathering structured feedback on CLI interactions to improve its intuitiveness, efficiency, and user satisfaction.
+
+The project covered several areas: assessing command organization and discoverability for both new and experienced users, reviewing help text quality across all commands, mapping common developer workflows to identify pain points, and evaluating error handling and user guidance during failures. But rather than just producing a report with recommendations, I worked on actively fixing the issues I discovered.
+
+## Understanding the User Perspective
+
+To better understand the challenges users face, I reached out to my peers in college to learn about their initial experience with Knative Functions. Their responses revealed a common pattern. Most people would start by reading the [func README](https://github.com/knative/func/blob/main/README.md){:target="_blank"}, then head to the [Knative documentation website](https://knative.dev/docs/functions/){:target="_blank"} for functions. The installation usually went smoothly without any issues. But then came what I found most fascinating: the curiosity phase.
+
+After installation, most people, including myself when I first started, would simply type commands like `func deploy`, `func build`, or `func invoke` just to see what happens. We were curious, exploring, trying to understand what each command does. But the problem was what they encountered. Error messages like this:
+
+```
+Error: '/mnt/c/Users/RAYYAN/Desktop/func' does not contain an initialized function
+```
+
+Technically correct, but not very helpful if you're just starting out. What should you do next? Where do you even begin?
+
+## Improving Help Text
+
+My initial weeks focused on improving the help text for all commands when run outside a function directory. I wanted to make those cryptic error messages actually helpful. 
+
+Here's how that same error looks now:
```

Comment:
@RayyanSeliya maybe make it a bit clearer what you mean by "that same error". Perhaps something like "Here's how that same error I mentioned above looks now with the new help text"